beginner_source/flava_finetuning_tutorial.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+"""
+TorchMultimodal Tutorial: Finetuning FLAVA
+============================================
+"""
+
+######################################################################
+# Multimodal AI has recently become very popular owing to its ubiquitous
+# nature, from use cases like image captioning and visual search to more
+# recent applications like image generation from text. **TorchMultimodal
+# is a library powered by Pytorch consisting of building blocks and end to
+# end examples, aiming to enable and accelerate research in
+# multimodality**.
+# 
+# In this tutorial, we will demonstrate how to use a **pretrained SoTA
+# model called** `FLAVA <https://arxiv.org/pdf/2112.04482.pdf>`__ **from
+# TorchMultimodal library to finetune on a multimodal task i.e. visual
+# question answering** (VQA).
+# 
+
+
+######################################################################
+# Installations
+#
+# We will use TextVQA dataset and bert tokenizer from HuggingFace for this
+# tutorial. So you need to install datasets and transformers in addition to TorchMultimodal.
+# When running this tutorial in Google Colab, install the required packages
+# by uncommenting the following:
+#
+"""
+!pip install torchmultimodal-nightly
+!pip install datasets
+!pip install transformers
+"""
+
+######################################################################
+# For this tutorial, we treat VQA as a classification task. So we need to
+# download the vocab file with answer classes and create the answer to
+# label mapping.
+#
+# We also load the `textvqa
+# dataset <https://arxiv.org/pdf/1904.08920.pdf>`__ from HuggingFace
+#
+
+with open("data/vocabs/answers_textvqa_more_than_1.txt") as f:
+  vocab = f.readlines()
+
+answer_to_idx = {}
+for idx, entry in enumerate(vocab):
+  answer_to_idx[entry.strip("\n")] = idx
+
+
+######################################################################
+# We see there are 3997 answer classes including a class representing
+# unknown answers
+#
+
+print(len(vocab))
+print(vocab[:5])
+
+from datasets import load_dataset
+dataset = load_dataset("textvqa")
+
+from IPython.display import display, Image
+idx = 5
+print("Question: ", dataset["train"][idx]["question"])
+print("Answers: " ,dataset["train"][idx]["answers"])
+display(dataset["train"][idx]["image"].resize((500,500)))
```

Comment:
Need to find a way to render this image correctly

Reply:
@malfet any suggestions on how to make the image render correctly. Right now this is what I see if I run it locally:

![Screen Shot 2022-09-27 at 2 55 07 PM (2)](https://user-images.githubusercontent.com/5317992/192884394-07023b37-2347-42f6-bcd9-4aebe915751f.png)

Reply:
This worked but someone might have a better solution:
```
from IPython.display import display, Image
import matplotlib.pyplot as plt
import numpy as np

idx = 5
print("Question: ", dataset["train"][idx]["question"])
print("Answers: " ,dataset["train"][idx]["answers"])
im = np.asarray(dataset["train"][idx]["image"].resize((500,500)))
plt.imshow(im)
plt.show()
```
![Screen Shot 2022-09-29 at 9 50 45 AM (2)](https://user-images.githubusercontent.com/5317992/193091473-ca33345b-69a9-4781-8510-26ab81b5cc5f.png)

Reply:
I'm a bit unsure, how you've rendered it? But traditionally we use matplotlib rather than IPython